experimental/packages/opentelemetry-instrumentation/src/index.ts

```diff
@@ -15,9 +15,8 @@
  */
 
 export * from './autoLoader';
-export { InstrumentationBase } from './platform/index';
-export { InstrumentationNodeModuleDefinition } from './instrumentationNodeModuleDefinition';
-export { InstrumentationNodeModuleFile } from './instrumentationNodeModuleFile';
+// Exports for Node.js and Browser are different, therefore exporting *
+export * from './platform/index';
```

Comment:
This is not the way the platform-specific exports work and will cause things to break

Reply:
@dyladan I see. We actually rely on this quite a bit though in other packages in this repo (exporter packages, mostly), so we should likely also get on top of this there. It is also (in spirit) the way we had it before it was before we merged #4336.

From what I can tell there are only three other ways for us to fix this then
- provide and maintain our own `normalize` implementation
- rely on a third-party package that provides an agnostic implementation
- normalize in Node.js while using a no-op in the browser (this does not seem like a safe option) 

WDYT?

Reply:
Before #4336 we actually had the bad situation I'm trying to avoid and I had to fix it to get that PR to pass.

>normalize in Node.js while using a no-op in the browser (this does not seem like a safe option)

Why do you think this is unsafe? There is no valid reason to use a NodeModuleFile in browser anyway and it would be completely broken in other ways. The only thing we need to fix is that we shouldn't import `path` in browser contexts.

Reply:
Looking closer into this, I agree that it's likely fine if we do that (no-op in browser). Closing this PR, will re-open another one with that approach :+1: